pkg/pipelines/dryrun.go

```diff
@@ -0,0 +1,62 @@
+package pipelines
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+)
+
+const scriptTemplate = `#!/bin/sh
+is_argocd=false
+argo_path="config/argocd"
+cicd_path="config/{{ .CICDEnv }}"
+client={{ .Client }}
+if [ -d ${argo_path} ]
+then
+   printf "Apply $(basename ${argo_path}) applications\n"
+   if [ "$client" != "" ];then $client apply --dry-run=$(inputs.params.DRYRUN) -k "${argo_path}/config"; fi
```

Comment:
I think I'd prefer to see us reject if `client == ""` in `makeScript` rather than having the test for it being all through the code.

It's not valid to not provide a command to execute?

Reply:
I understand why this is happening now, because of the param values.

I'd still prefer to see it as a function, rather than duplicating the check.

Reply:
Moved the redundant code to a function